src/pages/Login.tsx

```diff
@@ -58,10 +60,33 @@ const Login = () => {
     pwError,
   } = useLoginForm();
 
-  const handleLogin = () => {
+  const handleLogin = async () => {
     if (isValidId && isValidPw) {
-      login({ id, name: id });
-      navigate(from, { replace: true });
+      try {
+        const response = await axios.post('/api/login', {
+          email: id,
+          password: pw,
+        });
+        const token = response.data.data.authToken;
+        localStorage.setItem('authToken', token);
+        toast.success('로그인 성공!');
+        const userInfo = response.data.data;
+        login(userInfo);
+        navigate(from, { replace: true });
+      } catch (error: unknown) {
+        const axiousError = error as AxiosError<{ message?: string }>;
```

Comment:
```suggestion
        const axiosError = error as AxiosError<{ message?: string }>; // 😏
```

그리고 요기 assertion 문은 조금 더 안전하게 짤 수 있을거 같은데요! 

이 구문에서는 axios가 에러를 주로 던질테니까 Error 객체가 주로 넘어오겠지만.. JS 문법에서는 사실상 아무 값이나 throw가 가능합니다. `throw 1;`, `throw 'network error';` 이런식으로요. 그래서 typescript에서도 unknown 값으로 타입을 추론하는 것입니다. 

이런 지식을 기반으로 했을 때 아무값이나 올 수 있다고 가정한다면 동작은 똑같지만 아무래도 아래 코드가 조금 더 범용적이라고 볼 수 있을 거 같네요!

```ts
if (error instanceof Error && (axiosError.response.status >= 400 && axiosError.response.status < 500)) {
  console.log('Error 객체:', error.message);
} else  {
  console.log('network error');
}
```